### PR TITLE
K-allagbe/issue1-setup-repo

### LINKED ISSUE #1: Setup Repository for llamaindex-db
The `llamaindex-db` repository requires a comprehensive setup. 

### Tasks
- [x] Configure a `.devcontainer`.
- [x] Add relevant GitHub Actions workflows.
- [x] Add the initial codebase with comprehensive tests.
- [ ] #4
- [x] Add documentation.

### Acceptance Criteria
- DevContainer is configured with proper extensions.
- Relevant GitHub workflows should be present.
- Initial code should be present, well-organized, and thoroughly tested.
- The code should be organized in packages.
- The documentation is present.

### PR DESCRIPTION
**Pull request by k-allagbe**
Closes #1: repo setup and initial code

- [x] decontainer config
- [ ] workflows (partially)
- [x] initial code (notebook)
- [x] documentation

### CODE REVIEW

**Comment by k-allagbe** [comment]
@ThomasCardin please help me fix this: https://github.com/ai-cfia/llamaindex-db/actions/runs/7726927165/job/21064230103?pr=2. LICENSE file is present in this branch but not in `main` branch.

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+jupyterlab
+notebook
+asyncpg
+llama-index
+psycopg2
```
Comment (rngadam) [inline requirements.txt:5]: @leejaeka :

we're using psycopg3 in ailab-db : https://www.psycopg.org/ 

https://github.com/ai-cfia/ailab-db/blob/7db06b4a040f29ff11a65f6ab1d53b0c97b878ad/requirements.txt#L20

using psycopg2 should be deprecated as it is an older version.

**Review by rngadam — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (leejaeka) [reply requirements.txt:5]: psycopg2 was used to connect to postgresql. If psycopg3 does the same job, let us use the updated psycopg3

**Review by leejaeka — APPROVED** [review]
looks good to me

File: requirements.txt (hunk 1 above)
Comment (k-allagbe) [reply requirements.txt]: Done

**Review by rngadam — APPROVED** [review]

**Pull request merged**